apis/python/src/tiledbsoma/experimental/ingest.py

```diff
@@ -607,31 +798,70 @@ def _write_visium_spots(
     return soma_point_cloud
 
 
-def _write_visium_images(
-    image_pyramid: MultiscaleImage,
-    scale_factors: Dict[str, Any],
+def _create_or_open_scene(
+    uri: str,
     *,
-    input_hires: Union[None, str, Path],
-    input_lowres: Union[None, str, Path],
-    input_fullres: Union[None, str, Path],
     ingestion_params: IngestionParams,
+    context: Optional["SOMATileDBContext"],
+    additional_metadata: "AdditionalMetadata" = None,
+) -> Scene:
+    try:
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            scene = Scene.create(uri, context=context)
+    except (AlreadyExistsError, NotCreateableError):
+        # It already exists. Are we resuming?
+        if ingestion_params.error_if_already_exists:
+            raise SOMAError(f"{uri} already exists")
+        scene = Scene.open(uri, "w", context=context)
+
+    add_metadata(scene, additional_metadata)
+    return scene
+
+
+def _create_visium_tissue_images(
+    uri: str,
+    image_paths: List[Tuple[str, Path, Optional[float]]],
+    *,
     additional_metadata: "AdditionalMetadata" = None,
     platform_config: Optional["PlatformConfig"] = None,
     context: Optional["SOMATileDBContext"] = None,
+    ingestion_params: IngestionParams,
     use_relative_uri: Optional[bool] = None,
-) -> None:
-    # Write metadata from scale_factors file
-    for key, value in scale_factors.items():
-        image_pyramid.metadata[key] = value
-
-    # Add the different levels of zoom to the image pyramid.
-    for name, image_path in (
-        ("fullres", input_fullres),
-        ("hires", input_hires),
-        ("lowres", input_lowres),
-    ):
-        if image_path is None:
-            continue
+) -> MultiscaleImage:
+
+    # Open the first image to get the base size.
+    with Image.open(image_paths[0][1]) as im:
+        im_data_numpy = np.array(im)
+        ref_shape: Tuple[int, ...] = im_data_numpy.shape
+        im_data = pa.Tensor.from_numpy(im_data_numpy)
+
+    # Create the multiscale image.
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        image_pyramid = MultiscaleImage.create(
+            uri,
+            type=pa.uint8(),
+            reference_level_shape=ref_shape,
+            axis_names=("y", "x", "c"),
```

Comment:
reminder to double-check ordering of y,x,c

Reply:
Good catch. This was intentional for CZI, but we should revisit making an option.

Reply:
https://app.shortcut.com/tiledb-inc/story/56838